microraft-store-sqlite/src/main/java/io/microraft/store/sqlite/RaftSqliteStore.java

```diff
@@ -0,0 +1,346 @@
+    private static final Field<Integer> TERM = DSL.field("term", SQLDataType.INTEGER);
+    private static final Table<Record> LOG_ENTRIES = DSL.table("logEntries");
+    private static final Field<Long> INDEX = DSL.field("logIndex", SQLDataType.BIGINT);
+    private static final Table<Record> SNAPSHOT_CHUNKS = DSL.table("snapshotChunks");
+    private static final Field<Integer> CHUNK_INDEX = DSL.field("chunkIndex", SQLDataType.INTEGER);
+    private static final Table<Record> SNAPSHOTS = DSL.table("snapshots");
+    private static final Field<Integer> CHUNK_COUNT = DSL.field("chunkCount", SQLDataType.INTEGER);
+    private static final Name COUNT = DSL.name("count");
+
+    private final Field<RaftGroupMembersView> initialGroupMembersField;
+    private final Field<RaftEndpoint> localEndpointField;
+    private final Field<RaftEndpoint> votedForField;
+    private final Field<LogEntry> logEntryField;
+    private final Field<SnapshotChunk> chunkField;
+    private final CloseableDSLContext dsl;
+    private final RaftModelFactory raftModelFactory;
+    private boolean tryToCleanUpOldSnapshots = false;
+
+    public RaftSqliteStore(
+            CloseableDSLContext dsl, StoreModelSerializer modelSerializer, RaftModelFactory raftModelFactory) {
+        this.dsl = dsl;
+        this.raftModelFactory = raftModelFactory;
+        initialGroupMembersField = DSL.field(
+                "initialGroupMembers",
+                SQLDataType.BINARY.asConvertedDataType(new JooqConverterAdapter<>(
+                        modelSerializer.raftGroupMembersViewSerializer(), RaftGroupMembersView.class)));
+        logEntryField = DSL.field(
+                "logEntry",
+                SQLDataType.BINARY.asConvertedDataType(
+                        new JooqConverterAdapter<>(modelSerializer.logEntrySerializer(), LogEntry.class)));
+        localEndpointField = DSL.field(
+                "localEndpoint",
+                SQLDataType.BINARY.asConvertedDataType(
+                        new JooqConverterAdapter<>(modelSerializer.raftEndpointSerializer(), RaftEndpoint.class)));
+        votedForField = DSL.field(
+                "votedFor",
+                SQLDataType.BINARY.asConvertedDataType(
+                        new JooqConverterAdapter<>(modelSerializer.raftEndpointSerializer(), RaftEndpoint.class)));
+        chunkField = DSL.field(
+                "chunk",
+                SQLDataType.BINARY.asConvertedDataType(
+                        new JooqConverterAdapter<>(modelSerializer.snapshotChunkSerializer(), SnapshotChunk.class)));
+    }
+
+    private void createTablesIfNotExists() {
+        dsl.createTableIfNotExists(KV)
+                .column(KEY)
+                .primaryKey(KEY)
+                .column(localEndpointField)
+                .column(LOCAL_ENDPOINT_VOTING)
+                .column(initialGroupMembersField)
+                .column(TERM)
+                .column(votedForField)
+                .execute();
```

Comment:
this code is not compiling. I think `.primaryKey(KEY)` returns an intermediate object which has no `column(...)` method. The following code seems to be fine.

```
dsl.createTableIfNotExists(KV).column(KEY).column(localEndpointField).column(LOCAL_ENDPOINT_VOTING)
                .column(initialGroupMembersField).column(TERM).column(votedForField).primaryKey(KEY).execute();
```

Reply:
oops! I downgraded jooq because I realised that the version I was depending on required Java 17, that probably broke this.